test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorResolverTest.cs

```diff
@@ -41,6 +42,465 @@ private static TagHelperDescriptor Valid_InheritedTagHelperDescriptor
+                        new TagHelperDescriptor[0],
+                        new[]
+                        {
+                            new RazorError(
+                                string.Format(
+                                    invalidTagHelperPrefixValueError,
+                                    SyntaxConstants.CSharp.TagHelperPrefixKeyword,
+                                    '!',
+                                    "!"),
+                                directiveLocation1)
+                        }
+                    },
+                    {
+                        defaultAssemblyLookups,
+                        new[]
+                        {
+                            new TagHelperDirectiveDescriptor(
+                                "!th",
+                                directiveLocation1,
+                                TagHelperDirectiveType.TagHelperPrefix),
+                        },
+                        new TagHelperDescriptor[0],
+                        new[]
+                        {
+                            new RazorError(
+                                string.Format(
+                                    invalidTagHelperPrefixValueError,
+                                    SyntaxConstants.CSharp.TagHelperPrefixKeyword,
+                                    '!',
+                                    "!th"),
+                                directiveLocation1)
+                        }
+                    },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ResolveDirectiveDescriptorsInvalidTagHelperPrefixData))]
+        public void Resolve_CreatesExpectedErrorsForTagHelperPrefixDirectives(
+            Dictionary<string, IEnumerable<Type>> descriptorAssemblyLookups,
+            IEnumerable<TagHelperDirectiveDescriptor> directiveDescriptors,
+            IEnumerable<TagHelperDescriptor> expectedDescriptors,
+            IEnumerable<RazorError> expectedErrors)
+        {
+            // Arrange
+            var tagHelperDescriptorResolver =
+                new TestTagHelperDescriptorResolver(
+                    new LookupBasedTagHelperTypeResolver(descriptorAssemblyLookups));
+            var errorSink = new ParserErrorSink();
+            var resolutionContext = new TagHelperDescriptorResolutionContext(
+                directiveDescriptors,
+                errorSink);
+
+            // Act
+            var descriptors = tagHelperDescriptorResolver.Resolve(resolutionContext);
+
+            // Assert
+            Assert.Equal(expectedErrors, errorSink.Errors);
+            Assert.Equal(expectedDescriptors.Count(), descriptors.Count());
+
+            foreach (var expectedDescriptor in expectedDescriptors)
+            {
+                Assert.Contains(expectedDescriptor, descriptors, TagHelperDescriptorComparer.Default);
+            }
+        }
+
+        public static TheoryData ResolveDirectiveDescriptorsTagHelperPrefixData
```

Comment:
test a `TagHelperDirective` of type `TagHelperDirectiveType.TagHelperPrefix` with an empty prefix.  is that allowed or not?  can't tell from the tests in this class.